src/components/select/filterable-select/filterable-select.component.tsx

```diff
@@ -19,9 +20,70 @@ import useInputAccessibility from "../../../hooks/__internal__/useInputAccessibi
 
 let deprecateInputRefWarnTriggered = false;
 
-const FilterableSelectList = withFilter(SelectList);
+const FilterableSelectList = withFilter<SelectListProps>(SelectList);
 
-const FilterableSelect = React.forwardRef(
+export interface FilterableSelectProps
+  extends Omit<FormInputPropTypes, "defaultValue" | "value"> {
+  /** Prop to specify the aria-label attribute of the component input */
+  "aria-label"?: string;
+  /** Prop to specify the aria-labeledby property of the component input */
+  "aria-labelledby"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-component"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-element"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-role"?: string;
+  /** Child components (such as Option or OptionRow) for the SelectList */
+  children: React.ReactNode;
+  /** The default selected value(s), when the component is operating in uncontrolled mode */
+  defaultValue?: string | Record<string, unknown>;
```

Comment:
I don't know the component, will need to have a little investigation but is it not possible to avoid unknown here (and the few other places)? Would it not be `string | number` for example?

Reply:
yeah that's a good point. I'll be honest that I don't really know if it's needed or not - I deliberately adjusted a few of the newer places which didn't have types before so that the types matched the existing props (to avoid TS errors). I mostly didn't want to change the existing TS types (in the `.d.ts` file) to avoid breaking any consumers' implementations, but on reflection you're probably right that this probably shouldn't work (and likely won't work, in the sense of functioning correctly), if the record values are of some strange type.

Reply:
so having looked into this a bit more - although in practice consumers are most likely only going to want to use simple types like strings and numbers here, the idea is to allow them to use arbitrary data, so I'm not comfortable restricing this unless there's a very good reason. And I don't really see that. `unknown` is a bit like `any` in that anything can be assigned to it, but it's a more type-safe version that basically forces us to acknowledge in our code that we don't know what it is (unlike with `any` where we can do whatever we like, with the risk of things blowing up at runtime). This feels like the right balance to me both for our internal component code and for consumers - but I'm happy to be talked around if you have different ideas.

Reply:
actually on reflection this change isn't right, at least not on this place. The issue is that if `listActionButton` is something truthy which is still not a React Element - like a non-empty string - we want to just leave it as is, but the `React.isValidElement(listActionButton) && ...` will end up rendering nothing.

So I'm going to put this back to how I had it, as I think it's the only way to make all possible React Node types behave correctly here while also keeping TS happy. Let me know if you think I should add a test for this case and I will, but for now I'll keep it with the istanbul ignore comment.